cabal-install/Distribution/Client/CmdShowBuildInfo.hs

```diff
@@ -0,0 +1,240 @@
+                  installFlags mempty -- Not needed here
+                  haddockFlags
+                  testFlags
+
+-- Pretty nasty piecemeal out of json, but I can't see a way to retrieve output of the setupWrapper'd tasks
+showTargets :: Verbosity -> ProjectBaseContext -> ProjectBuildContext -> Lock -> IO ()
+showTargets verbosity baseCtx buildCtx lock = do
+  putStr "["
+  mapM_ showSeparated (zip [0..] targets)
+  putStrLn "]"
+    where configured = [p | InstallPlan.Configured p <- InstallPlan.toList (elaboratedPlanOriginal buildCtx)]
+          targets = fst <$> (Map.toList . targetsMap $ buildCtx)
+          doShowInfo unitId = showInfo verbosity baseCtx buildCtx lock configured unitId
+          showSeparated (idx, unitId)
+              | idx == length targets - 1 = doShowInfo unitId
+              | otherwise = doShowInfo unitId >> putStrLn "," 
+
+showInfo :: Verbosity -> ProjectBaseContext -> ProjectBuildContext -> Lock -> [ElaboratedConfiguredPackage] -> UnitId -> IO ()
+showInfo verbosity baseCtx buildCtx lock pkgs targetUnitId
+  | Nothing <- mbPkg = die' verbosity $ "No unit " ++ show targetUnitId 
+  | Just pkg <- mbPkg = do
+    let shared = elaboratedShared buildCtx
+        install = elaboratedPlanOriginal buildCtx
+        dirLayout = distDirLayout baseCtx
+        buildDir = distBuildDirectory dirLayout (elabDistDirParams shared pkg)
+        flags = setupHsBuildFlags pkg shared verbosity buildDir
+        args = setupHsBuildArgs pkg
+        srcDir = case (elabPkgSourceLocation pkg) of
+          LocalUnpackedPackage fp -> fp
+          _ -> ""
+        scriptOptions = setupHsScriptOptions 
+            (ReadyPackage pkg) 
+            install 
+            shared 
+            dirLayout
+            srcDir 
+            buildDir 
+            False 
+            lock
+        configureFlags = setupHsConfigureFlags (ReadyPackage pkg) shared verbosity buildDir
+        configureArgs = setupHsConfigureArgs pkg
+    --Configure the package if there's no existing config
+    lbi <- tryGetPersistBuildConfig buildDir
+    case lbi of
+      Left _ -> setupWrapper 
+                  verbosity 
+                  scriptOptions 
+                  (Just $ elabPkgDescription pkg) 
+                  (Cabal.configureCommand defaultProgramDb) 
+                  (const $ configureFlags)
+                  (const configureArgs)
+      Right _ -> pure ()
+    setupWrapper 
+      verbosity 
+      scriptOptions 
+      (Just $ elabPkgDescription pkg) 
+      (Cabal.showBuildInfoCommand defaultProgramDb) 
+      (const flags) 
+      (const args)
+    where mbPkg = find ((targetUnitId ==) . elabUnitId) pkgs
+
+-- | This defines what a 'TargetSelector' means for the @write-autogen-files@ command.
+-- It selects the 'AvailableTarget's that the 'TargetSelector' refers to,
+-- or otherwise classifies the problem.
+--
+-- For the @write-autogen-files@ command select all components except non-buildable and disabled
```

Comment:
Shouldn't this be moved to a separate PR as well if it concerns the `write-autogen-files` command?

Reply:
In my opiniion the comment might be outdated, since this function is neither exported nor used by `write-autogen-files`